source/ada/lsp-ada_handlers.adb

```diff
@@ -2022,6 +2056,219 @@ package body LSP.Ada_Handlers is
       return Response;
    end On_ALS_Called_By_Request;
 
+   ------------------------------
+   -- On_ALS_Calls_Request --
+   ------------------------------
+
+   overriding function On_ALS_Calls_Request
+     (Self    : access Message_Handler;
+      Request : LSP.Messages.Server_Requests.ALS_Calls_Request)
+      return LSP.Messages.Server_Responses.ALS_Calls_Response
+   is
+      use Libadalang.Analysis;
+
+      Value      : LSP.Messages.TextDocumentPositionParams renames
+        Request.params;
+      Response   : LSP.Messages.Server_Responses.ALS_Calls_Response
+        (Is_Error => False);
+      Imprecise  : Boolean := False;
+
+      procedure Process_Context (C : Context_Access);
+      --  Process the calls found in one context and append
+      --  them to Response.results.
+
+      procedure Add_Subprogram
+        (Subp : LSP.Messages.ALS_Subprogram_And_References);
+      --  Add a subprogram in results, it prevents having duplicates
+
+      ---------------------
+      -- Process_Context --
+      ---------------------
+
+      procedure Process_Context (C : Context_Access) is
+         Definition     : Defining_Name;
+         Calls          : LSP.Lal_Utils.References_By_Subprogram.Map;
+         Calls_Cursor   : LSP.Lal_Utils.References_By_Subprogram.Cursor;
+         This_Imprecise : Boolean;
+
+         function Process_Body_Children (N : Ada_Node'Class)
+                                         return Visit_Status;
+         --  Check if N is a call and if so resolve it and add it to a
+         --  map where the key is the Defining_Name of the call and the
+         --  value is a Double_Linked_List of Base_Id representing where
+         --  the call is made.
+
+         ----------------------------
+         -- Process_Body_Childreen --
+         ----------------------------
+
+         function Process_Body_Children (N : Ada_Node'Class)
+                                         return Visit_Status is
+         begin
+            --  Do not consider calls made by nested subprograms, expression
+            --  functions or tasks
+            Self.Trace.Trace (N.Image);
+            if N.Kind in
+              Ada_Subp_Body
+              | Ada_Subp_Spec
+              | Ada_Expr_Function
+
+              --  TODO: Reactivate these lines when libadalang supports
+              --  P_Next_Part for tasks: T716-049
+              --  | Ada_Task_Body
+              --  | Ada_Single_Task_Decl
+              --  | Ada_Task_Type_Decl
+            then
+               return Over;
+            end if;
+
+            if LSP.Lal_Utils.Check_If_Is_Call (N, Self.Trace, This_Imprecise)
```

Comment:
(style) the "then" should be at the end of the previous line, or, if it doesn't fit, it should look like

    if LSP.Lal_Utils.Check_If_Is_Call
      (N, Self.Trace, This_Imprecise)
    then